asQ/allatonce/form.py

```diff
@@ -59,6 +61,16 @@ def __init__(self,
 
         self.form = self._construct_form()
 
+    def time_update(self, t=None):
+
+        if t is not None:
+            self.t0.assign(t)
+
+        self.t0.assign(self.t0 + self.dt*self.aaofunc.ntimesteps)
```

Comment:
How come this has been taken outside the if-else block? Shouldn't we only do this if `t is None`?

Reply:
We still need to update the time series for both cases.

Reply:
Agreed, `time` needs updating from `t0` in both cases.
But here, if `t is not None` then `t0` is updated twice, so `t0!=t` at the end of this function.

Reply:
To be clear, I'm talking only about line 69 here, not the loop, which as you say should be the same no matter what `t` is.

Reply:
In this function, I am assuming that if t is not None, then it is the initial timestep of the previous window and it has to be shifted.